src/providers/EtherspotContextProvider.tsx

```diff
@@ -378,6 +379,38 @@ const EtherspotContextProvider = ({
     [sdk, accountAddress]
   );
 
+  const getRatesByTokenAddresses = useCallback(async (chainId: number, tokenAddresses: string[]) => {
+    let tokens  = tokenAddresses.filter((address) => address !== null)
+    if(!sdk || !tokens.length) return null
+    try {
+      const rates: RateData = await sdk.fetchExchangeRates({tokens, chainId});
+      if(!rates.errored && rates.items.length > 0) {
+        let addressByChain : Record<string, number> = {}
+        rates.items.forEach((rate) => {
+          addressByChain[rate.address] = rate.usd
+        })
+        return addressByChain
+      }
+    } catch (error) {
+      console.log("rateserror",error)
+    }
+
+  }, [sdk])
+
+  const getRatesByNativeChainId = useCallback(async (chainId: number) => {
+    if(!sdk) return null
+    try {
+      const rates: RateData = await sdk.fetchExchangeRates({tokens: ['0x0000000000000000000000000000000000000000'], chainId});
+      if(!rates.errored && rates.items.length > 0) {
```

Comment:
Same as above
```suggestion
      if (!rates?.errored && rates?.items?.length) {
```

Reply:
@Yash-Chaubal so we get this key everytime the requests succeed so rates.errored is always there if you see the Type RateData